store/badger.go

```diff
@@ -30,20 +40,25 @@ func NewDefaultInMemoryKVStore() KV {
 	}
 }
 
-func NewKVStore(rootDir, dbPath, dbName string, syncWrites bool) KV {
+func NewKVStore(ctx context.Context, rootDir, dbPath, dbName string, syncWrites bool, logger types.Logger) KV {
 	path := filepath.Join(Rootify(rootDir, dbPath), dbName)
-	db, err := badger.Open(badger.DefaultOptions(path).WithSyncWrites(syncWrites))
+	opts := memoryEfficientBadgerConfig(path, syncWrites)
+	db, err := badger.Open(*opts)
 	if err != nil {
 		panic(err)
 	}
-	return &BadgerKV{
+	b := &BadgerKV{
 		db: db,
 	}
+	if ctx != context.TODO() {
+		go b.gc(ctx, gcTimeout, discardRatio, logger)
```

Comment:
bit weird
can you comment?

Reply:
what exactl is the problem with running the gc even with context todo?
Cant you just handle ErrGCInMemoryMode at the callsite?

Reply:
i think it does not make sense. i actually removed the context.